lib/utils/mbaas-token-check.js

```diff
@@ -1,23 +1,31 @@
-var _ = require('underscore');
-var common = require("../common");
-var fhreq = require("./request");
+const _ = require('underscore');
+const common = require("../common");
+const fhreq = require("./request");
 
-module.exports = {
-  check: function(request, cb) {
-    var url = '/api/v2/mbaases/' + request.target;
-    common.doGetApiCall(fhreq.getFeedHenryUrl(), url, i18n._('Error checking MBaaS type'), function(err, response) {
-      if (err) {
-        return cb(err);
-      }
-      // Ensure that a token is provided if creating an openshift3 environment
-      if (response.type === 'openshift3' && typeof request.token === 'undefined') {
-        return cb(i18n._('OpenShift targets require a token to be provided'));
-      }
-      if (response.type === 'feedhenry' && request.token) {
-        return cb(i18n._('FeedHenry targets do not accept tokens'));
-      }
+function checkMbaasType(request, cb) {
+  const url = '/api/v2/mbaases/' + request.target;
+  common.doGetApiCall(fhreq.getFeedHenryUrl(), url, i18n._('Error checking MBaaS type'), function checkIfTokenRequired(err, response) {
+    if (err) {
+      return cb(err);
+    }
+
+    if (response.type === 'openshift3') {
+      request.targetType = 'openshift3';
+    }
+
+    // Ensure that a token is provided if creating an openshift3 environment
+    if (response.type === 'openshift3' && typeof request.token === 'undefined') {
+      return cb(i18n._('OpenShift targets require a token to be provided'));
+    }
+
+    if (response.type === 'feedhenry' && request.token) {
+      return cb(i18n._('FeedHenry targets do not accept tokens'));
+    }
 
-      return cb(null, request);
-    });
-  }
+    return cb(null, request);
+  });
+};
+
+module.exports = {
+  check: checkMbaasType
```

Comment:
This gets used as a check before [creating](https://github.com/feedhenry/fh-fhc/blob/master/lib/cmd/fh3/admin/environments/create.js#L40) or [updating](https://github.com/feedhenry/fh-fhc/blob/master/lib/cmd/fh3/admin/environments/update.js#L30) an environment. It ensures that if the environment that is being created is on an `openshift3` target that a token is provided. It turns out that there must be a `targetType` value in the request body to ensure supercore uses the `openshift3` handler to create the environment. I've just updated the check to append this `targetType` value if the MBaaS target is an `openshift3` one so now the environment project is created as expected.